tests/test_cleanup.py

```diff
@@ -0,0 +1,170 @@
+import logging
+import os
+from pathlib import Path
+import shutil
+import tempfile
+
+import pytest
+
+from orthoseg.lib import cleanup
+from orthoseg.helpers import config_helper as conf
+from tests import test_helper
+
+testprojects_dir = Path(tempfile.gettempdir()) / "orthoseg_test_cleanup"
+sampleprojects_dir = testprojects_dir / "sample_projects"
+footballfields_dir = sampleprojects_dir / "footballfields"
+
+
+def init_project():
+    shutil.rmtree(path=testprojects_dir, ignore_errors=True)
+    footballfields_dir.mkdir(parents=True, exist_ok=True)
+    shutil.copyfile(
+        src=test_helper.sampleprojects_dir / "project_template/projectfile.ini",
+        dst=footballfields_dir / "footballfields.ini",
+    )
+    shutil.copyfile(
+        src=test_helper.sampleprojects_dir / "imagelayers.ini",
+        dst=sampleprojects_dir / "imagelayers.ini",
+    )
+
+
+def load_project_config():
+    conf.read_orthoseg_config(
+        config_path=footballfields_dir / "footballfields.ini",
+        overrules=[
+            "general.segment_subject=footballfields",
+            "predict.image_layer=BEFL_2019",
+        ],
+    )
+
+
+@pytest.mark.parametrize(
+    "simulate, versions_to_retain, expected_files_in_dir",
+    [
+        (True, 3, 25),
+        (False, 3, 15),
+    ],
+)
+def test_cleanup_models(
+    simulate: bool, versions_to_retain: int, expected_files_in_dir: int
+):
+    # Init test project
+    init_project()
+    models_dir = footballfields_dir / "models"
+    models_dir.mkdir(parents=True, exist_ok=True)
+    models = [
+        "0.44293_0.hdf5",
+        "hyperparams.json",
+        "log.csv",
+        "model.json",
+        "report.pdf",
+    ]
+    for x in range(1, 6):
+        for model in models:
+            with open(models_dir / f"footballfields_0{x}_{model}", "w"):
+                pass
```

Comment:
```suggestion
            (models_dir / f"footballfields_0{x}_{model}").touch()
```

`Path` has a `touch()` function to create a new, empty file. Applicable to some more cases further on.